core/src/views/replica.rs

```diff
@@ -0,0 +1,67 @@
+use super::SystemView;
+use crate::nats::JetstreamSubscription;
+use crate::{messages::state::StateUpdate, nats::TypedNats, NeverResult};
+use anyhow::{anyhow, Result};
+use std::sync::{Arc, RwLock, RwLockReadGuard};
+use tokio::task::JoinHandle;
+
+pub struct SystemViewReplica {
+    view: Arc<RwLock<SystemView>>,
+    handle: JoinHandle<NeverResult>,
+}
+
+impl SystemViewReplica {
+    async fn snapshot_impl(
+        nats: TypedNats,
+    ) -> Result<(SystemView, JetstreamSubscription<StateUpdate>)> {
+        let mut view = SystemView::default();
+        let mut sub = nats.subscribe_jetstream::<StateUpdate>().await?;
+
+        if sub.has_pending {
+            while let Some((update, meta)) = sub.next().await {
+                view.update_state(update, meta.timestamp);
+
+                if !sub.has_pending {
+                    break;
+                }
+            }
```

Comment:
Does this read through all messages ever published to this stream to build the state?

Reply:
It reads through the most recent message published to each subject. It could also be constructed from a filtered stream (e.g. if we know in advance that we only want info about a specific drone), but I haven't implemented that yet.

Reply:
There's a subject for each backend, right? Does that mean this grows unbounded?